### ISSUE TITLE
Validation framework for model

### ISSUE THREAD

**Issue by sgreenbury** [Urban-Analytics-Technology-Platform/acbm#17]
Aim: define metrics to be used at different parts of the modelling to validate model against data. E.g. flows from QUANT model. See section in [wiki](https://github.com/Urban-Analytics-Technology-Platform/acbm/wiki/Adding-activity-patterns-to-synthetic-population#validation).

**Comment by sgreenbury** [comment]
Some measures to consider:
- Modelled travel times compared to reported travel times in NTS
- Modelled flows between MSOAs compared to observed flows used in QUANT

**Comment by Hussein-Mahfouz** [comment]
Guidelines from dft on activity and agent based models: [TAG unit M5-4](https://www.gov.uk/government/publications/tag-unit-m5-4-agent-based-methods-and-activity-based-demand-modelling)

**Comment by sgreenbury** [comment]
Notes on tasks:
- [ ] Retrieve data on MSOA to MSOA flows by mode
- [ ] Function to measure difference in NTS (ground truth) and estimated travel time (from travel time matrices)
- [ ] Calculate the aggregate MSOA to MSOA flows for a SPC matched population

**Comment by sgreenbury** [comment]
We also need to determine a set of metrics for measuring quality of matching between the two datasets as part of task1.

**Comment by BZ-BowenZhang** [comment]
> We also need to determine a set of metrics for measuring quality of matching between the two datasets as part of task1.

I completely agree with @sgreenbury, and I think calculating the metrics for comparison is easy. But after generating metrics, I have a question: How do we judge the 'goodness'? In other words, how do we set a threshold value as the acceptable standard? We do not have another candidate dataset to compare, but maybe we can compare it with other synthetic populations published in previous papers.

**Comment by sgreenbury** [comment]
Adding [reference](https://www.researchgate.net/publication/357268039_Activity-based_and_agent-based_Transport_model_of_Melbourne_AToM_an_open_multi-modal_transport_simulation_model_for_Greater_Melbourne) with validation methods from @stuartlynn

**Comment by BZ-BowenZhang** [comment]
- [ ]  Add more markdown descriptions for the notebook `Validation_SPC_with Cencus `

**Comment by BZ-BowenZhang** [comment]
As discussed on 20th Sep:
* The Census (commuting OD) dataset will be used as the benchmark dataset to validate the performance of different synthetic datasets (SPC and AcBM). The Census and SPC validation is completed and ready to merge with the main branch PR #50
*  For the next step, more validation work will be done after the whole Leeds population with activity chain is available, still comparing with the census data and comparing with the SPC.
*  In the further London case, we may have more option datasets to compare/validate such as mobile phone data from Tao's Group or travel flow from TfL etc.

**Comment by sgreenbury** [comment]
Closing this with merging of #37 #51 #84 #98. Future work (such as mentioned in [wiki](https://github.com/Urban-Analytics-Technology-Platform/acbm/wiki/%5B2%5D-Activity-Location-Assignment#32-validating-against-other-external-datasets-todo)) developing the validation framework further will be in a new issue.
